src/model/FanComponentModel.hpp

```diff
@@ -0,0 +1,296 @@
+
+    bool setFanWheelDiameter(double fanWheelDiameter);
+
+    bool setFanOutletArea(double fanOutletArea);
+
+    bool setMaximumFanStaticEfficiency(double maximumFanStaticEfficiency);
+
+    bool setEulerNumberatMaximumFanStaticEfficiency(double eulerNumberatMaximumFanStaticEfficiency);
+
+    bool setMaximumDimensionlessFanAirflow(double maximumDimensionlessFanAirflow);
+
+    bool setMotorFanPulleyRatio(double motorFanPulleyRatio);
+
+    void autosizeMotorFanPulleyRatio();
+
+    bool setBeltMaximumTorque(double beltMaximumTorque);
+
+    void autosizeBeltMaximumTorque();
+
+    bool setBeltSizingFactor(double beltSizingFactor);
+
+    bool setBeltFractionalTorqueTransition(double beltFractionalTorqueTransition);
+
+    bool setMotorMaximumSpeed(double motorMaximumSpeed);
+
+    bool setMaximumMotorOutputPower(double maximumMotorOutputPower);
+
+    void autosizeMaximumMotorOutputPower();
+
+    bool setMotorSizingFactor(double motorSizingFactor);
+
+    bool setMotorInAirstreamFraction(double motorInAirstreamFraction);
+
+    bool setVFDEfficiencyType(const std::string& vFDEfficiencyType);
+
+    bool setMaximumVFDOutputPower(double maximumVFDOutputPower);
+
+    void autosizeMaximumVFDOutputPower();
+
+    bool setVFDSizingFactor(double vFDSizingFactor);
+
+    bool setFanPressureRiseCurve(const Curve& bivariateFunctions);
+
+    bool setDuctStaticPressureResetCurve(const Curve& univariateFunctions);
+
+    bool setNormalizedFanStaticEfficiencyCurveNonStallRegion(const Curve& univariateFunctions);
+
+    bool setNormalizedFanStaticEfficiencyCurveStallRegion(const Curve& univariateFunctions);
+
+    bool setNormalizedDimensionlessAirflowCurveNonStallRegion(const Curve& univariateFunctions);
+
+    bool setNormalizedDimensionlessAirflowCurveStallRegion(const Curve& univariateFunctions);
+
+    bool setMaximumBeltEfficiencyCurve(const Curve& univariateFunctions);
+
+    void resetMaximumBeltEfficiencyCurve();
+
+    bool setNormalizedBeltEfficiencyCurveRegion1(const Curve& univariateFunctions);
+
+    void resetNormalizedBeltEfficiencyCurveRegion1();
+
+    bool setNormalizedBeltEfficiencyCurveRegion2(const Curve& univariateFunctions);
+
+    void resetNormalizedBeltEfficiencyCurveRegion2();
+
+    bool setNormalizedBeltEfficiencyCurveRegion3(const Curve& univariateFunctions);
+
+    void resetNormalizedBeltEfficiencyCurveRegion3();
+
+    bool setMaximumMotorEfficiencyCurve(const Curve& univariateFunctions);
+
+    void resetMaximumMotorEfficiencyCurve();
+
+    bool setNormalizedMotorEfficiencyCurve(const Curve& univariateFunctions);
+
+    void resetNormalizedMotorEfficiencyCurve();
+
+    bool setVFDEfficiencyCurve(const Curve& univariateFunctions);
+
+    void resetVFDEfficiencyCurve();
+
+    bool setEndUseSubcategory(const std::string& endUseSubcategory);
+
+    //@}
+    /** @name Other */
+    //@{
+
+    // Helper that creates defaulted Optional Curves
+    bool assignDefaultOptionalCurves();
```

Comment:
New helper to instantiate the optional curves

Reply:
I like that pattern